index.js

```diff
@@ -67,14 +67,15 @@ async function loadPlugins ({ pluginTree, options, opts, fastify }) {
 
 async function loadPlugin ({ file, type, directoryPrefix, options, log }) {
   const { options: overrideConfig, forceESM, encapsulate } = options
-  let content
+  let content, importedAsModule
   if (forceESM || type === 'module' || runtime.forceESM) {
     content = await import(pathToFileURL(file).href)
+    importedAsModule = true
   } else {
     content = require(file)
   }
 
-  if (isPluginOrModule(content) === false) {
+  if (isPluginOrModule(content, importedAsModule) === false) {
```

Comment:
actually... ist the following not enough?

```suggestion
  if (isPluginOrModule(content.default || content) === false) {
```

of maybe

```suggestion
  if (isPluginOrModule(importedAsModule ? content.default : content, importedAsModule) === false) {
```

Then you dont need any changes in `isPluginOrModule`

Reply:
thank you, I'd go with the second because it's clearer. I was too tunneling on fixing the function instead of the parameter.